File: DesignPatternsNew/java/StrategyDesignPattern/src/CSVExtraction.java
class CSVExtraction implements Extraction {

	public void extract() {
		//__________(1)___________
		open();

		//__________(2)___________
		System.out.println("I will extract by separating row by commas.");

		//__________(3)___________
		close();
	}

	public void open() {
		System.out.println("Opening CSV file");
	}

	public void close() {
		System.out.println("Closing CSV file");
	}
}

File: DesignPatternsExercises/ruby/strategy_design_pattern/solution/csv_extraction.rb
require_relative 'extraction'

class CSVExtraction < Extraction
  def extract
    # __________(2)__________
    open
    puts 'CSV Data type is getting extracted by separating row by commas...'
    # __________(3)__________
    close
  end

  def open
    puts 'Opening CSV file...'
  end

  def close
    puts 'Closing CSV file...'
  end
end
